Sources/MongoSwift/MongoCredential.swift

```diff
@@ -0,0 +1,75 @@
+/// Represents an authentication credential.
+public struct MongoCredential: Decodable, Equatable {
+    /// A string containing the username. For auth mechanisms that do not utilize a password, this may be the entire
+    /// `userinfo` token from the connection string.
+    public var username: String?
+
+    /// A string containing the password.
+    public var password: String?
+
+    /// A string containing the authentication database.
+    public var source: String?
+
+    /// The authentication mechanism. A nil value for this property indicates that a mechanism wasn't specified and
+    /// that mechanism negotiation is required.
+    public var mechanism: Mechanism?
+
+    /// A document containing mechanism-specific properties.
+    public var mechanismProperties: Document?
+
+    private enum CodingKeys: String, CodingKey {
+        case username, password, source, mechanism, mechanismProperties = "mechanism_properties"
+    }
+
+    public init(
+        username: String? = nil,
+        password: String? = nil,
+        source: String? = nil,
+        mechanism: Mechanism? = nil,
+        mechanismProperties: Document? = nil
+    ) {
+        self.username = username
+        self.password = password
+        self.source = source
+        self.mechanism = mechanism
+        self.mechanismProperties = mechanismProperties
+    }
+
+    /// Possible authentication mechanisms.
+    public struct Mechanism: Decodable, Equatable, CustomStringConvertible {
+        /// GSSAPI authentication.
+        /// - SeeAlso: https://docs.mongodb.com/manual/core/kerberos/
+        public static let gssAPI = Mechanism("GSSAPI")
```

Comment:
nit: we've decided to use `ID` for `ObjectID` and all of these cases use acronyms or initialism, they should probably mostly be uppercased

Reply:
Given that these are properties, I think they still have to conform to the lower camel case convention for the first chunk and can go full capitalization for subsequent chunks. (reference: https://github.com/raywenderlich/swift-style-guide/issues/101#issuecomment-320200304)

Reply:
feel free to ignore this comment, I don't have strong opinions and left an approval